### PR TITLE
[DOC] Updated docs: small fixes

### PR DESCRIPTION
**Pull request by outoftardis**
# Description

- Replaced Unicode character "‖" with "\\|"
- Replaced sphinx prompts with code-blocks (needed for Intel website output)
- Fixed bibliography references

### CODE REVIEW

File: docs/source/dev_guide/algorithms/implicit_als/implicit-alternating-least-squares.rst [hunk 1]
```diff
@@ -54,7 +54,7 @@ The ALS model is trained using the implicit ALS algorithm
 by minimizing the following cost function:
 
 .. math::
-	\underset{{x}_{*},{y}_{*}}{\mathrm{min}}\underset{u,i}{\mathrm{\Sigma }}{c}_{ui}{\left({p}_{ui}-{x}_{u}^{T}{y}_{i}\right)}^{2}+\lambda \left(\underset{u}{\mathrm{\Sigma }}{n}_{{x}_{u}}{‖{x}_{u}‖}^{2}+\underset{i}{\mathrm{\Sigma }}{m}_{{y}_{i}}{‖{y}_{i}‖}^{2}\right),
+	\underset{{x}_{*},{y}_{*}}{\mathrm{min}}\underset{u,i}{\mathrm{\Sigma }}{c}_{ui}{\left({p}_{ui}-{x}_{u}^{T}{y}_{i}\right)}^{2}+\lambda \left(\underset{u}{\mathrm{\Sigma }}{n}_{{x}_{u}}{\|{x}_{u}\|}^{2}+\underset{i}{\mathrm{\Sigma }}{m}_{{y}_{i}}{\|{y}_{i}\|}^{2}\right),
```
Comment (averbukh) [inline docs/source/dev_guide/algorithms/implicit_als/implicit-alternating-least-squares.rst:57]: Preview mode shows not correct result. ||xu|| and ||yi|| should be in the formulas.

**Review by averbukh — APPROVED** [review]

File: docs/source/dev_guide/algorithms/implicit_als/implicit-alternating-least-squares.rst (hunk 1 above)
Comment (outoftardis) [reply docs/source/dev_guide/algorithms/implicit_als/implicit-alternating-least-squares.rst:57]: This is the output for this equation, it's the same as the one published:
image

**Review by averbukh — APPROVED** [review]

**Pull request merged**